megamek/src/megamek/client/ui/swing/tileset/EntityImage.java

```diff
@@ -0,0 +1,405 @@
+    private static final File FILE_SMOKEFIRE_LRG = new File("units/DamageDecals", "SmokeFire3.png"); //$NON-NLS-1$
+    private static final File FILE_DAMAGEDECAL_EMPTY = new File("units/DamageDecals", "Transparent.png"); //$NON-NLS-1$
+    
+    private static Image dmgLight;
+    private static Image dmgModerate;
+    private static Image dmgHeavy;
+    private static Image dmgCrippled;
+    private static Image SmokeSml;
+    private static Image SmokeMed;
+    private static Image SmokeLrg;
+    private static Image SmokeFireSml;
+    private static Image SmokeFireMed;
+    private static Image SmokeFireLrg;
+    private static Image dmgEmpty;
+    private static boolean decalLoaded = false;
+
+    // Individual entity images
+    private Image base;
+    private Image wreck;
+    
+    /** A smaller icon used for the unit overview. */
+    private Image icon;
+    /** A color used instead of a camo. */
+    int tint;
+    private Image camo;
+    private Image[] facings = new Image[6];
+    private Image[] wreckFacings = new Image[6];
+    private Component parent;
+    /** The damage level, from none to crippled. */
+    private int dmgLevel;
+    /** The tonnage of the unit. */
+    private double weight;
+    /** True for units of class or subclass of Infantry. */
+    private boolean isInfantry;
+    /** True when the image is for an additional hex of multi-hex units. */
+    private boolean isSecondaryPos;
+    /** True when the image is for the lobby. */
+    private boolean isPreview;
+    /** True when the unit is likely to be more long than wide (e.g. tanks). */
+    private boolean isSlim;
+    /** True when the unit is likely to be very narrow (VTOL). */
+    private boolean isVerySlim;
+    
+
+    private final int IMG_WIDTH = HexTileset.HEX_W;
+    private final int IMG_HEIGHT = HexTileset.HEX_H;
+    private final int IMG_SIZE = IMG_WIDTH * IMG_HEIGHT;
+
+    public EntityImage(Image base, int tint, Image camo, Component comp, Entity entity) {
+        this(base, null, tint, camo, comp, entity, -1, true);
+    }
+    
+    public EntityImage(Image base, Image wreck, int tint, Image camo,
+            Component comp, Entity entity, int secondaryPos) {
+        this(base, wreck, tint, camo, comp, entity, secondaryPos, false);
+    }
+    
+    public EntityImage(Image base, Image wreck, int tint, Image camo,
+            Component comp, Entity entity, int secondaryPos, boolean preview) {
+        this.base = base;
+        this.tint = tint;
+        this.camo = camo;
+        parent = comp;
+        this.wreck = wreck;
+        this.dmgLevel = entity.getDamageLevel();
+        this.weight = entity.getWeight();
+        isInfantry = entity instanceof Infantry;
+        isSecondaryPos = secondaryPos != 0 && secondaryPos != -1;
+        isPreview = preview;
+        isSlim = entity instanceof Tank || entity instanceof Aero;

```

Comment:
Is there a way of determining this based on the unit image (without any too crazy computations)? e.g. checking width/height?  A Demolisher tank might warrant three smoke stacks, while a Maultier APC only really needs one.

Reply:
Possible by checking the pixels in the center row. I just think that it wouldn't make a big difference; two or three smoke stacks on a unit that could hold three but burns alright with two - in other words, all the pixel-counting effort wouldn't really show in the result. I wouldnt really want to do that to increase the number of smoke stacks. If on the other hand many units showed smoke stacks where there shouldnt be any and a weight test doesnt help, then that's be a good solution.

Reply:
I'm not aware of too many skinny 100-ton tanks. I guess the Long Tom + Trailer currently? Probably let's leave it for a later PR.